src/CSharpMongoMigrations/DatabaseMigrations.cs

```diff
@@ -23,10 +23,10 @@ internal class DatabaseMigrations : IDatabaseMigrations
         private readonly IMongoDatabase _db;
         private readonly string _collectionName = typeof(MigrationVersion).Name;
 
-        public DatabaseMigrations(string server, string database)
+        public DatabaseMigrations(MongoUrl url)
         {
-            var client = MongoClientFactory.Get($"mongodb://{server}");
-            _db = client.GetDatabase(database);
+            var client = new MongoClient(url);
```

Comment:
The idea of MongoClientFactory is reuse the one MongoClient for all migrations. 
It increases speed and performance drastically.
It's not good to instantiate MongoClient every time when you need to apply a migration.
You definitely need to restore it.